pkg/plugins/removerplugin.go

```diff
@@ -8,6 +8,39 @@ import (
 	"sigs.k8s.io/kustomize/api/resource"
 	"sigs.k8s.io/kustomize/kyaml/kio"
 	kyaml "sigs.k8s.io/kustomize/kyaml/yaml"
+
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
+)
+
+var (
+	WhitelistedFields = []RemoverPlugin{
+		// for resources, i.e cpu and memory
+		{
+			Gvk:  gvk.Deployment,
+			Path: []string{"spec", "template", "spec", "containers", "*", "resources"},
+		},
+		// for replicas
+		{
+			Gvk:  gvk.Deployment,
+			Path: []string{"spec", "replicas"},
+		},
+	}
+
+	// List of components should be whitelisted as not reconcile on WhitelistedFields.
+	WhitelistedComponent = map[string]*[]RemoverPlugin{
+		"codeflare":                       &WhitelistedFields,
+		"dashboard":                       &WhitelistedFields, // dashboard is a upstream component name
+		"rhods-dashboard":                 &WhitelistedFields, // rhods-dashboard is a downstream component name
```

Comment:
Is this needed? Can we pass component name from `updateResource` function to the plugin?

Reply:
Currently all the coponents have opted in for this plugin, do we still need to maintain this list?

Alternatively we can add support for annotation/label if users want operator to reconcile resources?

Reply:
i had the thought if we should by default just go with all component and skip this list.
but i feel maybe better to keep it for now with ability to opt-in and opt-out, e.g a new component maybe be added in the future but it is not ready for GA. and we can have a different logic on it, as exclude from the list till GA. (just one example).

annotation/label can be used in this case, but then it will be depending on the outcome of the "labeling"/"annotation" e.g `if found.GetAnnotations()[annotations.ManagedByODHOperator] == "false" && componentName == "kserve" {` is a hardcode will be eventually go away along with annotation

Reply:
Yes I think that should be okay. The default behaviour would be all the replicas and resources are configurabled for a component. If a user wants to opt-out just add annotation like `annotations.ManagedByODHOperator==true`. We need to provide a dynamic way for users to get operator defaults.

Reply:
updated